src/senaite/core/api/dtime.py

```diff
@@ -269,3 +272,100 @@ def to_iso_format(dt):
         DT = to_DT(dt)
         return to_iso_format(DT)
     return None
+
+
+def date_to_string(dt, fmt="%Y-%m-%d", default=""):
+    """Format the date to string
+    """
+    if not is_date(dt):
+        return default
```

Comment:
`is_date(dt)` evaluates to `True` if `dt` is a string. In such case, `dt.strftime(fmt)` won't fail because of year is before 1900, but because `dt` is not a date, datetime or DateTime object. I think we should consider the removal of the following check and conversion from `to_date` function:

https://github.com/senaite/senaite.core/blob/1da8e42eaf578c4a30cf950fb08667dfc95a2825/src/senaite/core/api/dtime.py#L61-L63

Otherwise, I suggest to do a safe conversion when `is_date` evaluates to `True`, as follows:

```python
if not is_date(dt):
    return default
dt = toDT(dt)
...
```

Reply:
Well spotted! I'll take the latter approach for now.

Reply:
ok, done